src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/AtmosphericDrag.cpp

```diff
@@ -95,23 +98,33 @@ SatelliteSystem AtmosphericDrag::getSatelliteSystem() const
     return satelliteSystem_;
 }
 
-void AtmosphericDrag::applyContribution(
-    const NumericalSolver::StateVector& x, NumericalSolver::StateVector& dxdt, const Instant& anInstant
+Array<Shared<const CoordinatesSubset>> AtmosphericDrag::getReadCoordinatesSubsets() const
+{
+    return {CartesianPosition::Default(), CartesianVelocity::Default()};
+}
+
+Array<Shared<const CoordinatesSubset>> AtmosphericDrag::getWriteCoordinatesSubsets() const
+{
+    return {CartesianVelocity::Default()};
+}
+
+VectorXd AtmosphericDrag::computeContribution(
+    const Instant& anInstant, const VectorXd& x, const Shared<const Frame>& aFrame
```

Comment:
Passing pointer by reference and naming convention for shared pointers

```suggestion
    const Instant& anInstant, const VectorXd& x, const Shared<const Frame> aFrameSPtr
```